### ISSUE TITLE
new_lle branch noPerspective Issues

### ISSUE THREAD

**Issue by m4xw** [gonetz/GLideN64#2181]
Hey @gonetz, have been checking out the new_lle branch.

This change causes all glDrawArrays calls to crash in libEGL on my Switch Port (changed it at all occurences):
![image](https://user-images.githubusercontent.com/13141469/75160636-ac515900-571a-11ea-9b89-7a88dba89025.png)

Its GL 4.3 Core, I assume (can't check rn) that noPerspective isnt populated (since it breaks too if i force it to false, since it takes the !GLESX codepath).

It works on my desktop setup without issues but I guess noPerspective is available there.

tldr: Issues with !GLESX and !noPerspective

**Comment by loganmc10** [comment]
noperspective is supposed to be available on OpenGL 3.0+

If the Switch really doesn't support it for some reason, opengl_GLInfo.cpp will probably need to be modified from:

```
noPerspective = Utils::isExtensionSupported(*this, "GL_NV_shader_noperspective_interpolation");
```
to something like:
```
noPerspective = (Utils::isExtensionSupported(*this, "GL_NV_shader_noperspective_interpolation") || !isGLESX) && !Switch;
```
And figure out some way to identify the Switch based on the GPU ( @fzurita has done this with the mobile GPUs like you can see here: https://github.com/gonetz/GLideN64/blob/master/src/Graphics/OpenGLContext/opengl_GLInfo.cpp#L46-L60)

And then the check can change from ```if(!_glinfo.isGLESX || _glinfo.noPerspective)``` to just checking for ```_glinfo.noPerspective```

**Comment by m4xw** [comment]
Guess I need to double check extension detection for that platform (fwiw we are forced to use novueau for homebrew)

**Comment by loganmc10** [comment]
@m4xw you might want to try the latest commit to new_lle to see if this issue is fixed. On Intel I had a shader linking issue because of the way the noperspective variable was declared. It's fixed now, it's possible the Shield had a similar issue

**Comment by m4xw** [comment]
Started my rebase, will give feedback once I added all the new fancy options and have it up and running.

**Comment by m4xw** [comment]
Seems resolved
